Tests/Realm.Tests/TestHelpers.cs

```diff
@@ -345,6 +345,21 @@ public static void DrainQueue<T>(this ConcurrentQueue<StrongBox<T>> queue, Actio
             }
         }
 
+        public static void DrainQueueAsync<T>(this ConcurrentQueue<StrongBox<T>> queue, Func<T, Task> action)
+        {
+            AsyncContext.Run(async () =>
+            {
+                await Task.Run(async () =>
```

Comment:
In this case, we don't really need the Nito context - we're just using it to make the Task.Run synchronous. I could also use `.Wait` instead of `AsyncContext.Run` - it's more of a habit than any practical consideration.